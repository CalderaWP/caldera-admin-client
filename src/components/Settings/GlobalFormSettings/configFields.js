import {optionFactory} from "../../util/optionFactory";

export const STYLE_FORM = 'SETTINGS/STYLE_INCLUDES/FORM';
export const STYLE_GRID = 'SETTINGS/STYLE_INCLUDES/FORM';
export const STYLE_ALERT = 'SETTINGS/STYLE_INCLUDES/STYLE_ALERT';
export const CDN_ENABLE = 'SETTINGS/CDN';
const enableOption = optionFactory(
	true,
	'Enable'
);
export default  [
	{
		id: STYLE_FORM,
		label: 'Form Styles',
		desc: 'Includes Bootstrap 3 styles on the frontend for form fields and buttons',
		type: 'checkbox',
		path: 'styleIncludes.form',
		options: [enableOption]
	},
	{
		id: STYLE_ALERT,
		label: 'Alert Styles',
		desc: 'Includes Bootstrap 3 styles on the frontend for form alert notices',
		type: 'checkbox',
		path: 'styleIncludes.alert',
		options: [enableOption]
	},
	{
		id: STYLE_GRID,
		label: 'Grid',
		desc: 'Includes Bootstrap 3 styles on the frontend for form grid layouts',
		type: 'checkbox',
		path: 'styleIncludes.grid',
		options: [enableOption]
	},{
		id: CDN_ENABLE,
		label: 'Enable Free CDN',
		desc: 'Some usage data will be shared with CDN providers',
		type: 'checkbox',
		path: 'otherSettings.cdnEnable',
		options: [enableOption]
	},

]