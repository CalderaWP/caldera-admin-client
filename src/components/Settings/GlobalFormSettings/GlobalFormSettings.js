import PropTypes from 'prop-types';
import configFields, {CDN_ENABLE, STYLE_ALERT, STYLE_FORM, STYLE_GRID} from './configFields'
import {object, pick} from "dot-object";
import {SettingsGroup} from "../SettingsGroup";

/**
 * Default values for style includes settings
 * @type {{[p: string]: boolean}}
 */
const styleIncludesDefaults = {

	[STYLE_FORM]: true,
	[STYLE_ALERT]: true,
	[STYLE_GRID]: true,

};

/**
 * Default values for other settings
 * @type {{[p: string]: boolean}}
 */
const otherSettingsDefaults = {
	[CDN_ENABLE]: false
};

/**
 * Create the global form settings UI
 * @param {Object} props
 * @return {*}
 * @constructor
 */
export class GlobalFormSettings extends SettingsGroup {
	constructor(props) {
		super(props);
		this.state = {
			styleIncludes: {
				...styleIncludesDefaults,
				...props.styleIncludes
			},
			otherSettings: {
				...otherSettingsDefaults,
				...props.otherSettings
			}
		}
	}

	/**
	 * Save the settings
	 */
	onSettingsSave() {
		this.props.onSettingsSave(
			this.state
		);
	}

	/**
	 * Update internal state when settings change
	 *
	 * @param {Object} update
	 */
	onSettingsChange(update) {
		this.setState(update);
	}


	/**
	 * Prepare config fields
	 *
	 * @return {Array}
	 */
	getConfigFields() {
		let currentConfigFields = this.props.configFields;
		currentConfigFields.forEach(configField => {
			const {path} = configField;
			configField.value = pick(
				path,
				this.state
			);

			configField.onValueChange = (newValue) => {
				const update = {
					...this.state,
					[path]: newValue
				};
				this.onSettingsChange(object(update));
			};

		});
		return currentConfigFields;
	};
};

/**
 * Prop types for the GlobalForms settings component
 * @type {{}}
 */
GlobalFormSettings.propTypes = {
	classNames: PropTypes.string,
	onSettingsSave: PropTypes.func.isRequired,
	styleIncludes: PropTypes.object,
	otherSettings: PropTypes.object
};
/**
 * Class names used in the GlobalForms settings component
 * @type {{wrapper: string}}
 */
GlobalFormSettings.classNames = {
	wrapper: 'caldera-forms-global-form-settings'
};

/**
 * Default props for the GlobalForms settings component
 * @type {{}}
 */
GlobalFormSettings.defaultProps = {
	configFields,
	styleIncludes: styleIncludesDefaults,
	otherSettings: otherSettingsDefaults,
	wrapperClass: GlobalFormSettings.classNames.wrapper
};



