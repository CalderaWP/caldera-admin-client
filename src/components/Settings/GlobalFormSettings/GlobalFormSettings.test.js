import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {GlobalFormSettings} from "./GlobalFormSettings";

Enzyme.configure({ adapter: new Adapter() });

describe( 'GlobalFormSettings component', () => {
	it( 'Matches snapshot with minimal props', () => {
		expect(
			renderer.create(
				<GlobalFormSettings
					onSettingsSave={() => {} }
				/>
			).toJSON()
		).toMatchSnapshot()
	});

	it( 'Is wrapped in the right class', () => {
		expect(
			shallow(
				<GlobalFormSettings
					onSettingsSave={() => {} }
				/>
			).find( '.' + GlobalFormSettings.classNames.wrapper )
				.length
		).toEqual(1)
	});
});