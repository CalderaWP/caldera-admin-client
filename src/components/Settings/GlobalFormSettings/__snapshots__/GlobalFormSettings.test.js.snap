// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFormSettings component Matches snapshot with minimal props 1`] = `
<div
  className="caldera-forms-global-form-settings"
>
  GlobalFormSettings
</div>
`;
